drivers/SmartThings/philips-hue/src/disco.lua

```diff
@@ -222,6 +250,20 @@ function HueDiscovery.search_bridge_for_supported_devices(driver, api_instance,
       end
     end
   end
+
+  if do_delete then
+    for _, device in ipairs(driver:get_devices()) do ---@cast device HueDevice
+      local device_type = device:get_field(Fields.DEVICE_TYPE)
+      local not_known_to_bridge = device_is_joined_to_bridge[device:get_field(Fields.HUE_DEVICE_ID) or ""]
+      local parent_device_id = device.parent_device_id or device:get_field(Fields.PARENT_DEVICE_ID) or ""
+      local parent_bridge_device = driver:get_device_info(parent_device_id)
+      local is_child_of_bridge = parent_bridge_device and (parent_bridge_device:get_field(Fields.BRIDGE_ID) == bridge_id)
+      if device_type == "light" and is_child_of_bridge and not not_known_to_bridge then
```

Comment:
Just curious, why is it necessary to check that the device_type is a light? I know we only support lights now, but if in the future we supported other devices this seems like one more place that would need to get updated. Is it not enough to know the device is a child of the bridge and it's been deleted?

Reply:
I think you're right. I can adjust this.